bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+propagate quickly through the network before the upgrade is complete. The
+downside of this approach is that it would more than double the size of the
+legacy `node_announcement` and most of the information would be duplicated. It
+also makes gossip queries tricky since `node_announcement_2` uses a block-height
+based timestamp instead of the unix timestamp used by the legacy message.
+There also does not seem to be a big benefit in spreading the 
+`node_announcement_2` message quickly since the speed of propagation of the 
+`channel_announcement_2` and `channel_update_2` messages will still depend on a
+network wide upgrade.
+
+## Specification
+
+### Feature Bits
+
+Unlike the original set of gossip messages, it needs to be explicitly advertised
+that nodes are aware of the new gossip messages defined in this document at
+least until the whole network has upgraded to only using taproot gossip. We thus
+define a new feature bit, `option_taproot_gossip` to be included in the `init`
+message as well as the _old_ `node_announcement` message. Note that for all
+other feature bits with the `N` and `C` contexts, it can be assumed that those
+contexts will switch to the new `node_announcement_2` and
+`channel_announcement_2` messages for all feature bits _except_ for
+`option_taproot_gossip` which only makes sense in the context of the old
+`node_announcement` message and not the new `node_announcement_v2` message.
+The `option_taproot` feature bit can also be implied when using the new taproot 
+gossip messages. 
+
+| Bits  | Name                    | Description                                       | Context | Dependencies     |
+|-------|-------------------------|---------------------------------------------------|---------|------------------|
+| 32/33 | `option_taproot_gossip` | Nodes that understand the taproot gossip messages | IN      | `option_taproot` | 
+
+A node can be assumed to understand the new gossip v2 messages if:
+
+- They advertise `option_taproot_gossip` in a legacy `node_announcement` message
+  OR
+- They have broadcast one of the new gossip messages defined in this document.
+
+Advertisement of the `option_taproot_gossip` feature bit should be taken to 
+mean:
+
+- The node understands and is able to forward all the taproot gossip messages.
+- The node may be willing to open an announced taproot channel.
+
+### `open_channel` Extra Requirements
+
+These extra requirements only apply if the `option_taproot` channel type is set
+in the `open_channel` message.
+
+The sender:
+
+- if `option_taproot_gossip` was negotiated:
+    - MAY set the `announce_channel` bit in `channel_flags`
+- otherwise:
+    - MUST NOT set the `announce_channel` bit.
+
+The receiving node MUST fail the channel if:
+
+- if `option_taproot_gossip` was not negotiated and the `announce_channel`
+  bit in the `channel_flags` was set.
+
+### `channel_ready` Extensions
+
+These extensions only apply if the `option_taproot` channel type was set in the
```

Comment:
Should specify that these values only need to be attached if an announcement signatures message hasn't already been full constructed. 

We also need to think about the restart implications here as well. Eg: lnd today will sign then stash the `ann_sig` message to read from disk to send to the remote party. However if each time we reconnect a new set of nonces is exchanged, then we'll need to wait for `chan_reest` to be sent (`channel_ready` isn't always sent once a channel is fully operational and has had updates), then we'll need to be ready to regen+ sign again.